cloud/ec2_client.go

```diff
@@ -204,6 +204,11 @@ func (c *awsClientImpl) RunInstances(ctx context.Context, input *ec2.RunInstance
 						return false, EC2InsufficientCapacityError
 					}
 					grip.Debug(message.WrapError(ec2err, msg))
+					grip.Debug(message.Fields{
```

Comment:
I think if there's an EC2InsufficientCapacityError it would be caught on 203, causing the caller to return false and that error on 204. This means that the debug statement on 206 won't get hit. However, I think RunInstances is unlikely to return interesting capacity errors, since I believe RunInstances is only used to get on-demand hosts.

Reply:
I would think so too, except I can find errors using the search string "InsufficientInstanceCapacity", but not "increment subnet" and "got EC2InsufficientCapacityError", which should show up in the calling function upon return of EC2InsufficientCapacityError (https://github.com/evergreen-ci/evergreen/blob/8f1a1622b544421afa3f29fb7d80376101bb28c8/cloud/ec2.go#L319). Furthermore, the errors that do show up with that search string match the fields for that debug error on 206/196, namely "api_name", "args", and "client".

Reply:
Consequently, I think the if statement is not being entered regularly.

Reply:
The InsufficientInstanceCapacity capacity errors are from CreateFleet, not RunInstances. I think they're coming from here https://github.com/evergreen-ci/evergreen/blob/cee32b0cd9309a29118784e14d28b35a785fb391/cloud/ec2_client.go#L984.

Reply:
That makes sense. I didn't think to search for the api_name, but I agree with you now. I think we can abandon/ close this PR.